Source/themes.cpp

```diff
@@ -523,17 +521,16 @@ void PlaceThemeMonsts(int t, int f)
  */
 void Theme_Barrel(int t)
 {
-	int xp, yp, r;
+	int xp, yp;
 	char barrnd[4] = { 2, 6, 4, 8 };
 	char monstrnd[4] = { 5, 7, 3, 9 };
 
 	for (yp = 0; yp < MAXDUNY; yp++) {
 		for (xp = 0; xp < MAXDUNX; xp++) {
 			if (dTransVal[xp][yp] == themes[t].ttval && !nSolidTable[dPiece[xp][yp]]) {
 				if (random_(0, barrnd[leveltype - 1]) == 0) {
-					if (random_(0, barrnd[leveltype - 1]) == 0) {
-						r = OBJ_BARREL;
-					} else {
+					_object_id r = OBJ_BARREL;
+					if (random_(0, barrnd[leveltype - 1]) != 0) {

```

Comment:
Can we use a ternary instead? Otherwise this causes double initialization, which adds a bit of cognitive overhead.

Reply:
I would prefer not to as the line already contains multiple operations and would exceed 80 columns.

Reply:
One good formatting approach with ternaries is to do this:

```c++
var x = thisIsAVeryLongConditionOhMyGodHowCanAnyoneWriteSuchBigComplicatedThings
    ? option1
    : option2;
```

If that's idiomatic C++ maybe we could adopt it? It mostly fixes the problem you were seeing.

Reply:
No, it fixes the minor part, I still don't like to have more than one operation per line.

Reply:
My main issue is with the complexity of the condition, your suggestion while better still has me not liking the many statements on the line (it's still a declaration and assignment longer).